### PR TITLE
#161319955 - Series endpoints for performing CRUD operations

### PR DESCRIPTION
**Pull request by macphilips**
#### What does this PR do?
Adds the feature that allows an author to create a series and add articles to the series.
#### Description of Task to be completed?
- Added test suites for series endpoints.
- Create endpoints for creating, updating, delete, and getting authors series
- Added pagination to GET series endpoint.
- Added pagination of articles when getting single series.
- Added pagination to GET comment endpoint.
- Added pagination of sub-comments when getting a single comment.
- Update PUT article endpoint to add existing article to a series.
- Add support for tagging a series.
- Add support for sending a notification when a new article is added to a series.
- Add support for sending notification to an author when a user follows a series.
- Updated and refactored commentRoute.test.js, searchArticles.test.js, profile.test.js files.
- Refactored routes that involve pagination, in order to have consistent pagination
- Corrected typos in Project files
- Updated Swagger documentation
#### How should this be manually tested?
- Fetch and checkout to the branch
```$xslt
> git fetch
> git checkout ft-161319955-create-articles-series 
> npm install 
```
- Setup database using .env-sample
* Run Sequelize migration 
```$xslt
 > npm run migration
 ```
- Run application
```$xslt 
> npm run start
```
- To run test
```$xslt 
> npm run test
```

* Using a REST client such as postman to test endpoints: 
To create series: `POST /api/v1/series` 
To get list of series: `GET /api/v1/series`, for paginated result:`GET /api/v1/series?page=1&size=5`
To update, get or delete a specific series:
-- `PUT /api/v1/series/:slug` 
-- `GET /api/v1/series/:slug` for paginated result:`GET /api/v1/series/:slug?page=1&size=5`.
-- `DELETE /api/v1/series/:slug`.
To get a list of series the logged in author has published: `/api/v1/me/series`

sample request body
```JSON
{
    "title": "My awesome series",
    "imageUrl": "http://localhost.com/link/to/my/awesome/series/cover.jpeg",
    "description": "Interdum et malesuada fames ac ante ipsum primis in faucibus."
}
```
* For API Documentation
http://localhost:3000/api-docs/

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- [#161319955](<https://www.pivotaltracker.com/story/show/161319955>)
#### Questions

### CODE REVIEW

File: tests/routes/series.route.test.js [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+ * @param {User} user2
+ * @return {Promise<any[]>} Creates and returns two dummy series.
+ */
+async function createDummySeries(user1, user2) {
+  const [series1, series2, articles1, articles2] = await Promise.all([
+    Series.create({
+      ...seriesContent,
+      slug: Util.createSlug('Ever wonder how?')
+    })
+      .then(s => s.setUser(user1)),
```
Comment (omedale) [inline tests/routes/series.route.test.js]: change variable name

File: controllers/series/FollowSeriesController.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * Crud controller for Series entity
+ */
+export default class FollowSeriesController {
+  /**
+   *
+   * @param {request} req
+   * @param {response} res
+   * @param {next} next
+   * @return {Promise<void>} r
```
Comment (Alameen688) [inline controllers/series/FollowSeriesController.js]: I think you forgot to write the complete return message here. You can have something like "returns a promise....",  'r' isn't descriptive.

File: controllers/series/FollowSeriesController.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+   *
+   * @param {number} userId - id of the user to follow the series
+   * @param {Series} series - series to follow or unfollow
+   * @param {'follow'|'unfollow'} follow - status of the follow event
+   * @return {Promise<void>} Create or update follow series relationship
+   */
+  static async follow(userId, series, follow) {
+    let status = follow.toUpperCase();
+    if (follow === 'unfollow') {
+      status = 'UN_FOLLOW';
```
Comment (Alameen688) [inline controllers/series/FollowSeriesController.js]: Nice job transforming to uppercase. However, UNFOLLOW is one word, you don't need to separate Un from Follow. By doing this, you will remove the need for the if statement.

**Comment by olusoladavid** [comment]
You have done an awesome job here @macphilips . You practically cleaned up and regularized many parts of the codebase. This is appreciated. I have tested the article series feature you implemented in this pull request. It works as requested in the acceptance criteria of the associated story card. Your code adheres strictly to the team's conventions too. I approve it for merging.

**Review by olusoladavid — APPROVED** [review]
You have done an awesome job here @macphilips . You practically cleaned up and regularized many parts of the codebase. This is appreciated. I have tested the article series feature you implemented in this pull request. It works as requested in the acceptance criteria of the associated story card. Your code adheres strictly to the team's conventions too. I approve it for merging.

**Pull request merged**